shared/src/lib.rs

```diff
@@ -7,7 +7,8 @@ use std::collections::hash_map::DefaultHasher;
 
 type Title = String;
 type Url = String;
-pub type PostId = (Title, Url);
+type Meta = Option<String>;
+pub type PostId = (Title, Url, Meta);
```

Comment:
Technically the metadata is not part of the post id. I wonder if we should introduce a new struct
```rust
pub struct PostKey {
  id: PostId,
  meta: Meta,
};
```

we could use it like so:
```rust
let key = PostKey { id: (post.title, post.url), meta: post.meta };
prepared.insert(key, post.body);
```

Reply:
Trying that generated other errors that I am unfamiliar with (being new to rust), once I get some time I can try and figure it out.

Reply:
I don't think it's a big deal to be honest. It's not 100% semantically correct since the meta field is not part of the post id, but it shouldn't break anything as far as I can see. So if you can't find a more idiomatic solution it's okay and we can merge it like that. 😉

Reply:
OK I will fix the test case then later today.